File: app/src/main/java/com/github/overpass/gather/ui/meeting/chat/users/list/viewholder/PendingUserViewHolder.java
package com.github.overpass.gather.ui.meeting.chat.users.list.viewholder;

import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.github.overpass.gather.R;
import com.github.overpass.gather.ui.meeting.chat.users.list.UsersAdapter;
import com.github.overpass.gather.ui.meeting.chat.users.model.UserModel;

import butterknife.BindView;

public class PendingUserViewHolder extends UserViewHolder {

    @BindView(R.id.flMenuContainer)
    FrameLayout flMenuContainer;
    @BindView(R.id.tvAccept)
    TextView tvAccept;

    public PendingUserViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    @Override
    public void bind(UserModel user, UsersAdapter.OnItemClickListener listener) {
        super.bind(user, listener);
        tvAccept.setOnClickListener(view -> listener.onItemClick(user.getId()));
    }

    public void updateMenu() {
        ViewGroup.LayoutParams params = flMenuContainer.getLayoutParams();
        params.height = itemView.getHeight();
        flMenuContainer.setLayoutParams(params);
    }
}

File: app/src/main/java/com/github/overpass/gather/screen/meeting/chat/users/list/viewholder/PendingUserViewHolder.kt
package com.github.overpass.gather.screen.meeting.chat.users.list.viewholder

import android.view.View
import android.widget.FrameLayout
import android.widget.TextView
import com.github.overpass.gather.screen.meeting.chat.users.list.UsersAdapter
import com.github.overpass.gather.screen.meeting.chat.users.model.UserModel
import kotlinx.android.synthetic.main.item_pending_user.view.*

class PendingUserViewHolder(itemView: View) : UserViewHolder(itemView) {

    private val flMenuContainer: FrameLayout = itemView.flMenuContainer
    private val tvAccept: TextView = itemView.tvAccept

    override fun bind(item: UserModel, listener: UsersAdapter.OnItemClickListener) {
        super.bind(item, listener)
        tvAccept.setOnClickListener { view -> listener.onItemClick(item.id) }
    }

    fun updateMenu() {
        val params = flMenuContainer.layoutParams
        params.height = itemView.height
        flMenuContainer.layoutParams = params
    }
}
